src/commands/deploy/getDeployFsPath.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { DialogResponses, IActionContext } from 'vscode-azureextensionui';
+import { configurationSettings, extensionPrefix } from '../../constants';
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { ext } from '../../extensionVariables';
+import { isPathEqual, isSubpath } from '../../utils/pathUtils';
+import * as workspaceUtil from '../../utils/workspace';
+import { getWorkspaceSetting, updateGlobalSetting } from '../../vsCodeConfig/settings';
+
+export async function getDeployFsPath(context: IActionContext, target: vscode.Uri | string | SiteTreeItem | undefined, fileExtension?: string): Promise<string> {
```

Comment:
Also can this be a separate PR? It affects all languages, not just .NET Core - and it's mostly copied so it should be easy to get approved/merged.